### ISSUE TITLE
about the performance of benge-8k dataset reproduction

### ISSUE THREAD

**Issue by Vipermdl** [HSG-AIML/GDA#1]
Thank you for your novel insights and useful code. When I reproduced the segmentation performance of the BENGE-8k dataset, I found that the SLR Fine-tuning results I got were quite different from the performance mentioned in the paper. The performance using SatMAE was even lower than that of MAE. Below is the results of SatMAE segmentation and the results of SatMAE continual pre-training. 
**SatMAE segmentation result:**
![image](https://github.com/user-attachments/assets/c2911c93-6f95-4152-8457-48d907729dba)
**SatMAE continual pre-training result:**
![image](https://github.com/user-attachments/assets/79849c10-c817-4b76-b3c7-16c6d552e0ba)

Below are the configuration files for continual pre-training and Supervised segemntation. We sincerely appreciate if you can help point out our problems.

**Continual pre-training configuration:**
      task: mae
      seed: 0
      verbose: True
      wandb:
        mode: online
        fast_dev_run: False
        entity: dongliang
        project: low-rank-da-mae
        log_model: True
        experiment_dir: logs/sat_mae
        cache_dir: ./cache/
      data:
        datamodule: BENGEDataModule 
        modality: s1 #aerial #s1
        size: 6 #20
        root: data/ben-ge-8k 
        bands: null 
        num_classes: 256 # 15 #7 #45 # 10 # 11
        img_size: 224
        few_shot_k: null
        few_shot_seed: null
      model:
        name: sat_mae
        type: mae
        patch_size: 16
        mask_ratio: 0.75
        freeze_backbone: True
        pretrained: True
        input_res: 1
        fixed_output_size: 0
        use_mask_token: True
        adapter: True
        adapter_trainable: True
        norm_trainable: True
        only_bias_trainable: False
        only_scaler_trainable: False
        adapter_type: low-rank-scaling # lora
        adapter_scale: 1.0
        adapter_hidden_dim: 8
        adapter_shared: False  # same adapter weights for every attention block
        train_patch_embed: False
        patch_embed_adapter: False
        patch_embed_adapter_scale: 1
        train_cls_mask_tokens: True
        train_all_params: False  # overwrites all the above (freeze_backbone, train_patch_embed, adapter, adapter_shared)
        loss_on_all_patches: True
      knn:
        knn_eval: False
        knn_k: 5
      optim:
        lr: 0.0001
        batch_size: 32
        warmup_epochs: 5
        num_workers: 0
        min_steps: 10000
        max_steps: 25000
**Segmentation configuration:**
        task: segmentation
        continual_pretrain_run: it79d9mj # 6vivgdyb # pnhnusx6 # kc7u3y5s #6vivgdyb # kc7u3y5s # pnhnusx6
        seed: 2
        verbose: True
        wandb:
          mode: online
          fast_dev_run: False
          entity: dongliang
          project: low-rank-da-satmae-seg
          log_model: True
          experiment_dir: logs/sat_mae_seg
          cache_dir: cache/
        data:
          datamodule: BENGEDataModule #EuroSATDataModule
          modality: s1
          root: data/ben-ge-8k # data/
          bands: null #[B01, B02, B03, B04, B05, B06, B07, B08, B08A, B09, B11, B12] # [B04,B03,B02]
          num_classes: 11 # 10
          img_size: 224
          few_shot_k: null
          few_shot_seed: null
        model:
          name: fcn # upernet
          backbone: sat_mae
          backbone_type: ""
          feature_map_indices: [5, 11, 17, 23]
          patch_size: 16
          freeze_backbone: True
          pretrained: True
          input_res: 10
          fixed_output_size: 0
          embed_dim: 1024
          use_mask_token: True
          adapter: True
          adapter_trainable: True
          norm_trainable: True
          adapter_type: low-rank-scaling
          adapter_scale: 1.0
          adapter_hidden_dim: 8
          adapter_shared: False  # same adapter weights for every attention block
          train_patch_embed: False
          patch_embed_adapter: False
          patch_embed_adapter_scale: 1
          train_cls_mask_tokens: False
          train_all_params: False  # overwrites all the above (freeze_backbone, train_patch_embed, adapter, adapter_shared)
          loss_on_all_patches: True
          only_scaler_trainable: False
          only_bias_trainable: False
        knn:
          knn_eval: False
          knn_k: 5
        optim:
          lr: 0.000001
          batch_size: 32
          warmup_epochs: 5
          num_workers: 0
          min_steps: 10000
          max_steps: 25000
          aux_loss_factor: 0.1
          lr_schedule_patience: 10

Thank you very much for handling this issue. We sincerely appreciate your time and look forward to your response.

**Comment by scheibenreif** [comment]
Hi @Vipermdl,

thanks for detailed report. Without checking your configuration too closely - is it possible that you are looking only at the performance on the validation set? Confusingly, the final `test_*` metrics on wandb also corresponds to the performance on the validation set. To test the resulting model on the true test set, you can use the `main_test_checkpoint.py` script from the repo. Simply create a file similar to `configs/run_logs/run_ids.csv` with your wandb run id, dataset name, training setting (lin eval or ft), and model name as columns and point the main of `main_test_checkpoint.py` to the location of that file. I have updated the README to reflect all this better.

Best,

Linus

**Comment by Vipermdl** [comment]
Thanks for your detailed reply, which helped us a lot. According to your suggestions, we reproduced the performance on the benge-8k segmentation task as follows. 
![image](https://github.com/user-attachments/assets/8b6ae274-28e5-4fa7-8695-4d39cd9ec09b)
Is it close to the results in Table 1 in the paper (69.9 vs 70.63)? We are not sure whether the accuracy used in Table 1 is Jaccard Index or other metrics. The paper does not seem to mention the definition of metrics.  

Best regards,
dongliang

**Comment by scheibenreif** [comment]
Hi dongiang,

yes that seems very close (probably because of the different seed in the segmentation training). The table in the paper shows accuracy

Best,

Linus
